includes/Feed/FeedConfigurationDetection.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Feed;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use Error;
+use SkyVerge\WooCommerce\Facebook\Utilities\Heartbeat;
+use SkyVerge\WooCommerce\Facebook\Feed\FeedFileHandler;
+
+/**
+ * A class responsible detecting feed configuration.
+ */
+class FeedConfigurationDetection {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( Heartbeat::DAILY, array( $this, 'get_data_source_feed_tracker_info' ) );
```

Comment:
Note - this runs on daily heartbeat, so you will likely need to force the heartbeat to test.